libs/core/src/main/java/org/opensearch/core/xcontent/XContentBuilder.java

```diff
@@ -157,6 +157,9 @@ public static XContentBuilder builder(XContent xContent, Set<String> includes, S
 
     /**
      * Returns a string representation of the builder (only applicable for text based xcontent).
+     * Note: explicitly or implicitly (from debugger) calling toString() could cause XContentBuilder
```

Comment:
Thanks for the comment, @chishui , pushing it further: should we replace `toString` impl to not dump the content, the method is clearly dangerous to use.

Reply:
Ideally, we should, but if you search "builder.toString()", you'll find many implementation which simply `return builder.toString()` without closing it. There could be similar usage in plugins as well. Seems to be pretty big effort and could bring potential risk.

Reply:
Fair point, we may be indeed constrained by what we could do in 2.x without breaking things (updating javadocs is OK), but 3.x is certainly the place where  we may try to get rid of such dangerous API

Reply:
I think this documentation update is the right first step. I would also support changing the behavior in 3.x, though I'm not super concerned about this behavior. To me, asking to see the contents of a builder, even in toString(), is somewhat akin to "building" it and it's not crazy to expect the builder to no longer be mutable after that operation. But I get that it is unintuitive.